src/league.py

```diff
@@ -20,85 +20,94 @@ class League(object):
 
-        teams = resp.json()['teams']
+        return {
+            'current_week': data['status']['currentMatchupPeriod'],
+            'nba_day': data['status']['latestScoringPeriod'],
+            'members': members,
+            'teams': teams
+        }
 
-        # NOTE: This can be fixed up to be faster
-        team_id_data = [x for x in teams if x.get('id') == self.team_id]
+    def _fetch_team_meta(self, team_id=None):
 
-        logger.info(f'My Team Data: {json.dumps(team_id_data, indent=4)}')
+        team_id = team_id if team_id is not None else self.team_id
 
-    def _fetch_teams(self):
         params = {
             'view': 'mTeam'
         }
+        url = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
+        resp = requests.get(url=url, params=params, cookies=self.cookies)
 
-        req = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
-        resp = requests.get(req, params=params, cookies=self.cookies)
+        request_status(resp.status_code)
 
-        self.status = resp.status_code
-        request_status(self.status)
+        # raise error if bad response
+        teams = resp.json()['teams']
 
-        team_data = resp.json()
+        ## fantasy team ids start at 1 not 0
+        my_team = teams[team_id-1]
+
+        team_meta = {
+            'id': team_id,
+            'abbrev': my_team['abbrev'],
+            'owner': my_team['primaryOwner'],
+            'name': my_team['nickname'],
+            'record': (
+                my_team['record']['overall']['wins'],
+                my_team['record']['overall']['losses'],
+                my_team['record']['overall']['ties']
+            ),
+            'stats': my_team['valuesByStat']
+        }
 
-        return team_data
+        logger.info(f'My Team Data: {json.dumps(team_meta, indent=4)}')
 
-    def _fetch_roster(self):
+        return team_meta
+
+
+    def _fetch_rosters(self, team_id=None):
         params = {
             'view': 'mRoster',
-            'scoringPeriod': self.current_week
+            # 'scoringPeriod': self.current_week
         }
 
-        req = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
-        resp = requests.get(req, params=params, cookies=self.cookies)
+        url = f'{FBA_ENDPOINT}{self.year}/segments/0/leagues/{self.league_id}'
+        resp = requests.get(url=url, params=params, cookies=self.cookies)
+
+        request_status(resp.status_code)
 
-        self.status = resp.status_code
-        request_status(self.status)
+        data = resp.json()
 
-        rosters = resp.json()['teams']
+        ## make more efficient
+        rosters = {}
+        for team in data['teams']:
+            rosters[team['id']] = []
+            for player in team['roster']['entries']:
+                rosters[team['id']].append((
+                    player['playerId'],
+                    player['playerPoolEntry']['player']['fullName']
+                ))
```

Comment:
Why make this an array rather than 2 key value pairs?

Reply:
What do you mean?  The result becomes something like:

```

Reply:
Fair enough - I missed the tuple structure for `player_id` and `full_name` was thinking accessing these would get a bit wonky if it was an list of lists.

To be clear its up to preference but accessing (IMO) can be more explicit:
Up to you

```bash
{
    1: [
            {
                'id': 6583, 
                'fullName'': Anthony Davis'
            },
            {
                'id': 6450,
                'fullName': 'Kawhi Leonard'
            }
        ]
}
```